coding-the-legislation/35_periods.md

```diff
@@ -1,25 +1,103 @@
+
+
+## Periods for variable
+
 Most of the quantities calculated in openfisca can change over time. Therefore, each formula calculates a variable for a person (or a family, etc.) **for a given period**.
 
 This period is always the second argument of the formulas :
+
 ```py
+class salary(Variable):
+    column = FloatCol
+    entity = Person
+    label = u"Salary for a month"
+    definition_period = MONTH
+
     def function(person, period):
         ...
 ```
 
-A period can be a month, a year, `n` successive months or `n` successive years.
+The size of the period is constrained by the class attribute `definition_period` :
+  - `definition_period = MONTH` : The variable may have a different value each month. *For example*, the salary of a person. When `function` is executed, the parameter `period` will always be a whole month. Trying to compute `salary` with a period that is not a month will raise an error before entering `function`.
+  - `definition_period = YEAR` : The variable is defined for a year or it has always the same value every months of a year. *For example*, if taxes are to be paid yearly, the corresponding variable is yearly. When `function` is executed, the parameter `period` will always be a whole year (from January 1st to December 31th).
+  - `definition_period = ETERNITY` : The value of the variable is constant. *For example*, the date of birth of a person never changes. `period` is still the 2nd parameter of `function`. However when `function` is executed, the parameter `period` can be anything and it should not be used.
+
+
+## Calculating dependencies for a period different than the one they are defined for
+
+Calling a formula with a period that is incompatible with the attribute `definition_period` will cause an error. For instance, if we assume that a person `salary` is paid monthly:
+
+```py
+class taxes(Variable):
+    column = FloatCol
+    entity = Person
+    label = u"Taxes for a whole year"
+    definition_period = YEAR
+
+    def function(person, period):  # period is a year because definition_period = YEAR
+        salary_past_year = person('salary', period)  # salary is computed on a year while it's a montly variable, openfisca will complain
+        ...
+```
+
+However, sometimes, we do need to estimate a variable for a different period that the one it is defined for.
+
+We may for example want to get the sum of the salaries perceived on the past year, or the past 3 months. The option `ADD` tells openfisca to split the period into months, compute the variable for each month and sum up the results :
+
+```py
+class taxes(Variable):
+    column = FloatCol
+    entity = Person
+    label = u"Taxes for a whole year"
+    definition_period = YEAR
+
+    def function(person, period):  # period is a year because definition_period = YEAR
+        salary_last_year = person('salary', period, options = [ADD])
+        salary_last_3_months = person('salary', period.last_3_months, options = [ADD])
```

Comment:
This line doesn't really fit your example : `last_three_month` is not relevant for a year. Also, you haven't introduced `last_3_months` yet. I'd remove it.